packages/oas/src/operation/lib/get-example-groups.ts

```diff
@@ -0,0 +1,207 @@
+        // and it matches one of the existing example keys
+        if (mediaTypeExample.title && Object.keys(groups).includes(mediaTypeExample.title)) {
+          groups[mediaTypeExample.title].response = {
+            mediaType,
+            mediaTypeExample,
+            status: example.status,
+          };
+
+          // if the current group doesn't already have a name set,
+          // use the response example object's summary field
+          if (!groups[mediaTypeExample.title].name) {
+            groups[mediaTypeExample.title].name = mediaTypeExample.summary;
+          }
+        }
+      });
+    });
+  });
+}
+
+/**
+ * Returns a name for the given custom code sample. If there isn't already one defined,
+ * we construct a fallback value based on where the sample is in the array.
+ */
+function getDefaultName(sample: Extensions['code-samples'], count: number): string {
+  return sample.name && sample.name.length > 0 ? sample.name : `Default${count > 1 ? ` #${count}` : ''}`;
+}
+
+/**
+ * Returns an object with groups of all example definitions (body/header/query/path/response/etc.).
+ * The examples are grouped by their key when defined via the `examples` map.
+ *
+ * Any custom code samples defined via the `x-readme.code-samples` extension are returned,
+ * regardless of if they have a matching response example.
+ *
+ * For standard OAS request parameter (e.g., body/header/query/path/etc.) examples,
+ * they are only present in the return object if they have a corresponding response example
+ * (i.e., a response example with the same key in the `examples` map).
+ */
+export function getExampleGroups(operation: Operation): ExampleGroups {
+  let namelessCodeSamples = 0;
+  const groups: ExampleGroups = {};
+
+  // add custom code samples
+  const codeSamples = getExtension('code-samples', operation.api, operation) as Extensions['code-samples'][];
+  codeSamples?.forEach(sample => {
+    namelessCodeSamples += 1;
+    const name = getDefaultName(sample, namelessCodeSamples);
+
+    // sample contains `correspondingExample` key
+    if (groups[sample.correspondingExample]?.customCodeSamples?.length) {
+      groups[sample.correspondingExample].customCodeSamples.push(sample);
+    } else if (sample.correspondingExample) {
+      groups[sample.correspondingExample] = {
+        name,
+        customCodeSamples: [sample],
+      };
+    }
+
+    // sample does not contain a corresponding response example
+    else if (groups[noCorrespondingResponseKey]?.customCodeSamples?.length) {
+      groups[noCorrespondingResponseKey].customCodeSamples.push(sample);
+    } else {
+      groups[noCorrespondingResponseKey] = {
+        name,
+        customCodeSamples: [sample],
+      };
+    }
+  });
+
+  // if we added any custom code samples, add the corresponding response examples and return
+  if (Object.keys(groups).length) {
+    addMatchingResponseExamples(groups, operation);
+    return groups;
```

Comment:
how come we return here?

Reply:
if we have any custom code samples, we don't also want to add native OAS request examples. the custom code samples take precedence